internal/nlwrap/nlwrap_linux.go

```diff
@@ -56,32 +59,44 @@ func (h Handle) Close() {
 
 func retryOnIntr(f func() error) {
 	for attempt := 0; attempt < maxAttempts; attempt += 1 {
-		if err := f(); !errors.Is(err, unix.EINTR) {
+		if err := f(); !errors.Is(err, netlink.ErrDumpInterrupted) {
 			return
 		}
 	}
 	log.G(context.TODO()).Infof("netlink call interrupted after %d attempts", maxAttempts)
 }
 
-// AddrList calls nlh.LinkList, retrying if necessary.
+func discardErrDumpInterrupted(err error) error {
+	if errors.Is(err, netlink.ErrDumpInterrupted) {
+		// The netlink function has returned possibly-inconsistent data along with the
+		// error. Discard the error and return the data. This restores the behaviour of
+		// the netlink package prior to v1.2.1, in which NLM_F_DUMP_INTR was ignored in
+		// the netlink response.
+		log.G(context.TODO()).Warnf("discarding ErrDumpInterrupted: %+v", perrors.WithStack(err))
```

Comment:
Wondering if the stack trace really needs to be logged as a Warning. Since this was the old behavior, I think we'd know by now if it was really causing any issue. I'd rather log that with debug level.

Reply:
Maybe we wouldn't know though - "sometimes on my host" issues might have been written off as "can't repro"?

I think it's unlikely we'll see failures after five retries. But, if we do, it's probably something we want to know about. And, without a stack trace, there's no way to work out whether the error relates to any observed misbehaviour.

So, I think it's worth keeping the stack trace at info level - at least for 28.x, in case there is an issue in the real world?

Reply:
Fine, let's keep it as a Warning then.

Reply:
Do we need a tracking issue for updating it after 28.0?

Reply:
I don't think so ... either we'll see lots of errors with stack traces or, I think more likely, we won't see any.

If we don't see any, might as well leave it in - in case an environment where it's a problem crops up. If we do see the error happening a lot, it'll need addressing anyway.